django/chat/responses.py

```diff
@@ -89,27 +92,17 @@ def chat_response(chat, response_message, eval=False):
             content_type="text/event-stream",
         )
 
-    temperature = chat.options.chat_temperature
-    llm = AzureChatOpenAI(
-        azure_endpoint=settings.AZURE_OPENAI_ENDPOINT,
-        azure_deployment=f"{model}",
-        model=model,
-        api_version=settings.AZURE_OPENAI_VERSION,
-        api_key=settings.AZURE_OPENAI_KEY,
-        temperature=temperature,
-    )
-
-    if eval:
-        # Just return the full response and an empty list representing source nodes
-        return llm.invoke(chat_history).content, []
+    # TODO: Update eval stuff
+    # if eval:
+    #     # Just return the full response and an empty list representing source nodes
+    #     return llm.invoke(chat_history).content, []
 
-    llm_stream = llm.astream(chat_history)
     return StreamingHttpResponse(
         streaming_content=htmx_stream(
             chat,
             response_message.id,
-            response_stream=llm_stream,
-            chunk_property="content",
+            response_replacer=llm.chat_stream(chat_history),
+            llm=llm,
```

Comment:
The LLM object is passed in here because its costing methods need to be called *after* the response is complete. Otherwise there would be no reference to the object from which to access the token counts.